In [1]:
import pandas as pd;

In [2]:
revenue = pd.read_csv('monthly_revenue.csv')
subscriptions = pd.read_csv('subscriptions.csv')

In [28]:
subscriptions.head()

,customer_id,plan,billing_cycle,industry,company_size,seats,monthly_revenue,acquisition_channel,region,signup_date,churned,churn_date,churn_reason,support_tickets_12mo,nps_score,feature_usage_pct,upgraded
0,CUST-0001,Professional,Monthly,Retail,51-200,64,825.55,Organic Search,North America,2022-06-27,Yes,2024-03-05,Price Too High,13,3,15,No
1,CUST-0002,Professional,Annual,Retail,51-200,35,409.62,Direct Sales,North America,2023-03-19,No,NaN,NaN,1,3,52,Yes
2,CUST-0003,Starter,Annual,Finance,500+,266,1004.49,Partner,North America,2024-10-19,Yes,2025-04-13,Price Too High,10,4,31,No
3,CUST-0004,Business,Annual,Real Estate,11-50,34,1006.44,Partner,Europe,2023-01-07,No,NaN,NaN,3,7,58,No
4,CUST-0005,Starter,Annual,Manufacturing,51-200,48,198.43,Direct Sales,North America,2025-09-23,No,NaN,NaN,3,7,59,No


In [29]:
subscriptions.shape

(600, 17)

In [30]:
subscriptions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 600 entries, 0 to 599
Data columns (total 17 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   customer_id           600 non-null    object 
 1   plan                  600 non-null    object 
 2   billing_cycle         600 non-null    object 
 3   industry              600 non-null    object 
 4   company_size          600 non-null    object 
 5   seats                 600 non-null    int64  
 6   monthly_revenue       600 non-null    float64
 7   acquisition_channel   600 non-null    object 
 8   region                600 non-null    object 
 9   signup_date           600 non-null    object 
 10  churned               600 non-null    object 
 11  churn_date            313 non-null    object 
 12  churn_reason          313 non-null    object 
 13  support_tickets_12mo  600 non-null    int64  
 14  nps_score             600 non-null    int64  
 15  feature_usage_pct     6

In [31]:
subscriptions.isnull().sum()

customer_id               0
plan                      0
billing_cycle             0
industry                  0
company_size              0
seats                     0
monthly_revenue           0
acquisition_channel       0
region                    0
signup_date               0
churned                   0
churn_date              287
churn_reason            287
support_tickets_12mo      0
nps_score                 0
feature_usage_pct         0
upgraded                  0
dtype: int64

In [36]:
subscriptions.duplicated().sum()

np.int64(0)

In [52]:
subscriptions["plan"].unique().tolist()

['Professional', 'Starter', 'Business', 'Enterprise']

In [51]:
subscriptions["billing_cycle"].unique().tolist()

['Monthly', 'Annual']

In [35]:
revenue.head()

,month,total_active_customers,new_customers,churned_customers,monthly_churn_rate_pct,total_mrr,avg_revenue_per_customer,customer_acquisition_cost
0,2022-01,0,5,0,0.00,0.00,0.00,164.92
1,2022-02,5,10,1,20.00,7333.07,1466.61,189.21
2,2022-03,14,13,0,0.00,12301.16,878.65,139.49
3,2022-04,28,11,3,10.71,24798.76,885.67,259.09
4,2022-05,35,11,1,2.86,29091.88,831.20,152.96


In [37]:
revenue.shape

(48, 8)

In [38]:
revenue.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 8 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   month                      48 non-null     object 
 1   total_active_customers     48 non-null     int64  
 2   new_customers              48 non-null     int64  
 3   churned_customers          48 non-null     int64  
 4   monthly_churn_rate_pct     48 non-null     float64
 5   total_mrr                  48 non-null     float64
 6   avg_revenue_per_customer   48 non-null     float64
 7   customer_acquisition_cost  48 non-null     float64
dtypes: float64(4), int64(3), object(1)
memory usage: 3.1+ KB


In [39]:
revenue.isnull().sum()

month                        0
total_active_customers       0
new_customers                0
churned_customers            0
monthly_churn_rate_pct       0
total_mrr                    0
avg_revenue_per_customer     0
customer_acquisition_cost    0
dtype: int64

In [40]:
revenue.duplicated().sum()

np.int64(0)

## Revenue Dataset

Rows: 48  
Columns: 8

Each row is a statistic for the month and year, chronologically

No missing or duplicated values

**total_mrr:** Monthly Recurring Revenue (MRR) 

**customer_acquisition_cost:** 
* Customer Acquisition Cost (CAC)
* CAC = Marketing Spent / New Customers

## Subscriptions Dataset

Rows: 600  
Columns: 17

Each row represents a customer and their plan and billing cycle. Along with that, it has additional information such as signup date, churned date, company size, and more.

287 missing values in churn date and churn reason due to companies not leaving

**Billing cycle:**
* Monthly
* Annual

**Plans:**
* starter
* professional
* business
* enterprise

**support_tickets_12mo:** How many support requests were submitted during the last 12 months  

**nps_score (NPS):** How likely are you to recommend us?

Dataset inspection revealed no duplicate records and no significant data quality issues. Missing values in churn-related fields were expected for active customers.

# Basic EDA

In [41]:
subscriptions["churned"].value_counts()

churned
Yes    313
No     287
Name: count, dtype: int64

In [42]:
subscriptions["monthly_revenue"].describe()

count      600.000000
mean       817.654300
std       1352.584783
min         24.650000
25%        137.137500
50%        348.250000
75%        874.832500
max      10978.000000
Name: monthly_revenue, dtype: float64

In [53]:
subscriptions["plan"].value_counts()

plan
Starter         217
Professional    173
Business        160
Enterprise       50
Name: count, dtype: int64

In [54]:
subscriptions["billing_cycle"].value_counts()

billing_cycle
Monthly    352
Annual     248
Name: count, dtype: int64

In [ ]:
subscriptions["company_size"].value_counts()

company_size
11-50      176
1-10       157
51-200     141
201-500     88
500+        38
Name: count, dtype: int64

## Business Familiarization

**Plan:**
* Starter: 217
* Professional: 173
* Business: 160
* Enterprise: 50

**Billing cycle:**
* Monthly: 352
* Annual: 248

**Company size:**
* 1-10: 158
* 11-50: 176
* 51-200: 141
* 201-500: 88
* 500+: 38

**Churned:**
* Yes: 313
* No: 287

**Monthly revenue:**
* Mean: 818
* Median: 348
* Std: 1353